src/libraries/Microsoft.Extensions.Options/tests/SourceGenerationTests/Baselines/NetCoreApp/Validators.g.cs

```diff
@@ -0,0 +1,1756 @@
+            var context = new global::System.ComponentModel.DataAnnotations.ValidationContext(options);
+
+            context.MemberName = "Value";
+            context.DisplayName = baseName + "Value";
+            builder.AddResult(global::__OptionValidationStaticInstances.__Attributes.A5.GetValidationResult(options.Value, context));
+
+            return builder.Build();
+        }
+    }
+}
+#pragma warning disable CS0618 // Type or member is obsolete
+namespace Enumeration
+{
+    partial struct FirstValidator
+    {
+        /// <summary>
+        /// Validates a specific named options instance (or all when <paramref name="name"/> is <see langword="null" />).
+        /// </summary>
+        /// <param name="name">The name of the options instance being validated.</param>
+        /// <param name="options">The options instance.</param>
+        /// <returns>Validation result.</returns>
+        [global::System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.Extensions.Options.SourceGeneration", "42.42.42.42")]
+        public global::Microsoft.Extensions.Options.ValidateOptionsResult Validate(string? name, global::Enumeration.FirstModel options)
+        {
+            var baseName = (string.IsNullOrEmpty(name) ? "FirstModel" : name) + ".";
+            var builder = new global::Microsoft.Extensions.Options.ValidateOptionsResultBuilder();
+            var context = new global::System.ComponentModel.DataAnnotations.ValidationContext(options);
+
+            if (options.P1 != null)
+            {
+                var count = 0;
+                foreach (var o in options.P1)
+                {
+                    if (o is not null)
+                    {
+                        builder.AddResult(global::Enumeration.__SecondModelValidator__.Validate(baseName + $"P1[{count++}]", o));
+                    }
+                    else
+                    {
+                        builder.AddError(baseName + $"P1[{count++}] is null");
+                    }
+                }
+            }
+
+            if (options.P2 != null)
+            {
+                var count = 0;
+                foreach (var o in options.P2)
+                {
+                    if (o is not null)
+                    {
+                        builder.AddResult(global::__OptionValidationStaticInstances.__Validators.V2.Validate(baseName + $"P2[{count++}]", o));
+                    }
+                    else
+                    {
+                        builder.AddError(baseName + $"P2[{count++}] is null");
+                    }
+                }
+            }
+
+            if (options.P3 != null)
+            {
+                var count = 0;
+                foreach (var o in options.P3)
+                {
+                    if (o is not null)
+                    {
+                        builder.AddResult(global::Enumeration.__SecondModelValidator__.Validate(baseName + $"P3[{count++}]", o));
```

Comment:
There are some cases where `count` doesn't get incremented when the element is null, which will result in misleading error messages for anything after a null element.

Reply:
nice catch :-)